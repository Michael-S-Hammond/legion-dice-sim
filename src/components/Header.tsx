import '../css/Header.css';

import React from 'react';

class Header extends React.Component<any> {
    constructor(props : any) {
        super(props);
    }

    render() {
        return (
            <nav className="navbar navbar-dark bg-dark">
                <a className="navbar-brand" href="https://www.legionacademy.net/">
                    <img className="la-logo d-none d-sm-inline-block my-auto" alt="The Legion Academy Podcast"></img>
                </a>
                <h3 className="nav-header d-none d-sm-inline">The Legion Academy</h3>
                <h4 className="nav-header d-inline d-sm-none">The Legion Academy</h4>

                <button className="navbar-toggler black" type="button" data-toggle="collapse"
                        data-target="#navbarSupportedContent15" aria-controls="navbarSupportedContent15"
                        aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>

                <div className="collapse navbar-collapse" id="navbarSupportedContent15">
                    <ul className="navbar-nav mr-auto">
                        <li className="nav-item active">
                            <a className="nav-link" href="https://www.legionacademy.net/">Legion Academy - Home</a>
                        </li>
                        <li className="nav-item active">
                            <a className="nav-link" href="https://www.legionacademy.net/unit-analysis-and-mechanics">Legion Academy - Analysis</a>
                        </li>
                        <li className="nav-item active">
                            <a className="nav-link" href="https://www.legionacademy.net/terrain-studio">Terrain Studio</a>
                        </li>
                    </ul>
                </div>
            </nav>
        );
    }
}

export default Header;
