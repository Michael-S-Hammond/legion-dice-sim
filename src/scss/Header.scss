.la-logo {
    content: url('../img/LegionAcademyLogo.png');
    height: 50px;
    width: 50px;
    display: inline-block;
}

.nav-header {
    color: white;
}
