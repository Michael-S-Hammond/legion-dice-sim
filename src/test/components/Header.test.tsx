import React from 'react';

import { create } from 'react-test-renderer';

import Header from '../../components/Header';

describe('Header', () => {
    it('matches the snapshot', () => {
        const snapshot = create(<Header></Header>)
        expect(snapshot.toJSON()).toMatchSnapshot();
    });
});
