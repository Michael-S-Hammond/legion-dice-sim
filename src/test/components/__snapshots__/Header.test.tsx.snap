// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
<nav
  className="navbar navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="https://www.legionacademy.net/"
  >
    <img
      alt="The Legion Academy Podcast"
      className="la-logo d-none d-sm-inline-block my-auto"
    />
  </a>
  <h3
    className="nav-header d-none d-sm-inline"
  >
    The Legion Academy
  </h3>
  <h4
    className="nav-header d-inline d-sm-none"
  >
    The Legion Academy
  </h4>
  <button
    aria-controls="navbarSupportedContent15"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler black"
    data-target="#navbarSupportedContent15"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent15"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="https://www.legionacademy.net/"
        >
          Legion Academy - Home
        </a>
      </li>
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="https://www.legionacademy.net/unit-analysis-and-mechanics"
        >
          Legion Academy - Analysis
        </a>
      </li>
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="https://www.legionacademy.net/terrain-studio"
        >
          Terrain Studio
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
